torch/_dynamo/convert_frame.py

```diff
@@ -210,10 +211,20 @@ def _convert_frame_assert(frame: types.FrameType, cache_size: int, hooks: Hooks)
         increment_frame()
         code = frame.f_code
 
-        if code in input_codes and config.error_on_recompile:
-            raise exc.RecompileError(
-                f"Recompiled function {code.co_name} in {code.co_filename}"
+        if code in input_codes and (
+            recompilations_log.isEnabledFor(logging.DEBUG) or config.error_on_recompile
+        ):
+            message = (
+                f"Recompiling function {code.co_name} in {code.co_filename}",
+                f"triggered by the following guard failure: {str(guard_failures[code][-1])}",
```

Comment:
Are you sure this is correct? guard_failures is only recorded where we cooked out a reason, which only happens on guard_fail_hook on first, last, or if we plumb a guard_fail_fn.

So, you won't see all recompiles with this - it will misreport reasons.

Reply:
I tested this, this might be okay actually, as long as you don't want to see intermediary failures.